internal/controller/linodecluster_controller.go

```diff
@@ -154,7 +154,9 @@ func (r *LinodeClusterReconciler) reconcile(
 	// Handle deleted clusters
 	if !clusterScope.LinodeCluster.DeletionTimestamp.IsZero() {
 		if err := r.reconcileDelete(ctx, logger, clusterScope); err != nil {
-			if !reconciler.HasConditionSeverity(clusterScope.LinodeCluster, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
+			if !reconciler.HasStaleCondition(clusterScope.LinodeCluster,
+				clusterv1.ReadyCondition,
+				reconciler.DefaultTimeout(r.ReconcileTimeout, reconciler.DefaultClusterControllerReconcileTimeout)) {
 				logger.Info("re-queuing cluster/nb deletion")
 				return ctrl.Result{RequeueAfter: reconciler.DefaultClusterControllerReconcileDelay}, nil
```

Comment:
I wonder what our approach should be here. Currently we are doing:
```
if err != nil {
     hasStaleCondition() {
         # try delete every 5 seconds
         return {5 min}, nil
     }
     # try delete with exponential backoff
     return {...}, err
}
```
I wonder for delete, we should just have delete every 5 or 10 seconds and not have exponential backoff. With exponential backoffs, deletes could get stuck for 1000 secs (16+ mins) max if something goes wrong. Looking for thoughts from others.

Reply:
I was thinking about this yesterday. Do we know how many times does it have to fail to get stuck for 1000+ sec?

Reply:
This is what I have seen in past:
```
   0.004881143569946289
   0.022043943405151367
   0.04145383834838867
   0.08641910552978516
   0.1618349552154541
   0.3228580951690674
   0.6422019004821777
   1.2825469970703125
   2.561889171600342
   5.122460842132568
   10.242242097854614
   20.482059001922607
   40.962461948394775
   81.92205500602722
   0.0189058780670166
   163.84187817573547
   655.36194896698
   1000.0018429756165
   1000.0022790431976
   1000.0020320415497
   1000.0018348693848
```
So mostly failures 16-17 times. I see in most of the places we are doing exponential first and then serial when its a stale condition, I wonder if its a fixed pattern we want to use here as well or we want to have fixed for delete.

Reply:
I think it would be okay to do 10 or more seconds but I'm worried that a bad delete request could slow down the controllers from processing other requests Or could the controller handle that well?

Reply:
Since controllers pick from queue, a bad request would mean that request will keep on getting requeued every 10 secs if its using fixed requeue logic or exponentially if its using exponential logic. Since default concurrency is 10, I don't think it will slow down things much as its just for delete and we don't have too many deletes all the time. With exponential delete, I just see that someone might see their resources not getting cleaned up on UI fast if its stuck in exponential backoff. But that also is a rare case. I am ok with current approach, just wanted some extra eyes and thoughts here.

Reply:
I'm fine with doing 10 sec delayed requeue. Its a rare case but probably will happen :)